src/state/sagas/datasets/DeleteDataset/DeleteDataset.js

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { takeEvery, call, put, select } from 'redux-saga/effects';
+import { t } from 'i18next';
+import { Api } from '../../../../services/config/Api';
+import { dispatchSetApplicationErrorMessage } from '../../../dispatchers/app/ApplicationDispatcher';
+import { DATASET_ACTIONS_KEY } from '../../../commons/DatasetConstants';
+import { STATUSES } from '../../../commons/Constants';
+
+const getDatasets = (state) => state.dataset.list.data;
+const getSelectedDatasetIndex = (state) => state.dataset.selectedDatasetIndex?.data;
+
+export function* deleteDataset(action) {
+  const datasets = yield select(getDatasets);
+  const selectedDataset = yield select(getSelectedDatasetIndex);
+  const selectedDatasetId = datasets[selectedDataset]?.id;
```

Comment:
Maybe we should move this line inside the `try` block below, to prevent the app from crashing if the index of the selected dataset is invalid somehow?